arrow-data/src/main/kotlin/arrow/data/State.kt

```diff
@@ -37,22 +37,46 @@ typealias StateKindPartial<S> = StateTKindPartial<IdHK, S>
  */
 typealias State<S, A> = StateT<IdHK, S, A>
 
+/**
+ * tuple class to represent the result of a monadic computation (returned by `run` in the `State` monad)
+ *
+ * @param S type of the state
+ * @param A type of the returned value
+ */
+typealias StateResult<S, A> = Tuple2<S, A> // typealias necessary here because data classes can't be extended
+
+/**
+ * extracts the final state of a monadic computation in `State` out of the returned `StateResult`
+ *
+ * @param S type of the state
+ * @param A type of the returned value
+ */
+val <S,A> StateResult<S,A>.state: S get() = this.a
```

Comment:
This extension will leak to Tuple2, I believe. Can anyone please double-check?

Reply:
It will leak for sure if someone imports `import arrow.data.*`, we can leave the type alias where it is and the extensions to `import arrow.syntax.state.*` so it requires an implicit import. Thoguhts @tschuchortdev ?

Reply:
yeah, I think it will. But I don't really see a better solution since `Tuple2` is a data class that can't be extended, but is required in order to work with the `Applicative` functions. The only option is probably to move it to the syntax module like Raul said, so it has to be imported manually. 

I should also add a constructor function for `StateResult` with the right argument names.

Reply:
You could create a new type by wrapping the tuple and forwarding the accessor methods, if that was your intention. Is it worth creating this new object for the sake of the API?

Reply:
@pakoito I asked this return to remain a Tuple2 because we are building generic machinery for those. Creating a new type is fine if we add yet another extension to the result of `run()` that returns the state as a tuple for example `run().tupled()`